bin/libs/fs.sh

```diff
@@ -48,7 +48,25 @@ convert_to_absolute_path() {
 }
 
 get_tmp_dir() {
-  echo ${TMPDIR:-${TMP:-/tmp}}
+  print_error ">> Check if TMPDIR points to writable directory"
+  is_directory_writable "${TMPDIR}"
+  if [ $? -eq 0 ]; then
+    echo ${TMPDIR}
+  else
+    print_error ">> Check if TMP points to writable directory"
+    is_directory_writable "${TMP}"
+    if [ $? -eq 0 ]; then
+      echo ${TMP}
+    else
+      print_error ">> Check if /tmp is writable directory"
+      is_directory_writable "/tmp"
```

Comment:
Any way we can reduce repeating code? These lines show up 3 times in this function.

Reply:
done